prql-compiler/src/ast/pl/expr.rs

```diff
@@ -446,15 +446,15 @@ impl Expr {
             Error::new(Reason::Expected {
                 who: who.map(|s| s.to_string()),
                 expected: expected.to_string(),
-                found: format!("`{}`", Expr::from(i)),
+                found: format!("`{}`", Expr::new(i)),
             })
             .with_span(self.span)
         })
     }
 }
 
-impl From<ExprKind> for Expr {
-    fn from(kind: ExprKind) -> Self {
+impl Expr {
+    pub fn new(kind: ExprKind) -> Self {
```

Comment:
Makes sense. 

We used to use `.into()` and it does make sense because `kind` is the most important field and easily adding all other default fields was on-par with the idea of the from trait. But it was not obvious what `.into()` was doing so I started using `Expr::from` a lot. This PR is essentially saying "don't use `into` at all".